File: library-management-system/frontend/src/services/books.js
import api from "../api";

export const listBooks = () => api.get("/books").then((r) => r.data);
export const getBook = (id) => api.get(`/books/${id}`).then((r) => r.data);
export const createBook = (payload) =>
  api.post("/books", payload).then((r) => r.data);
export const updateBook = (id, payload) =>
  api.put(`/books/${id}`, payload).then((r) => r.data);
export const deleteBook = (id) => api.delete(`/books/${id}`).then((r) => r.data);

export const searchBooks = (params) =>
  api.get("/search", { params }).then((r) => r.data);

File: library-management-system/backend/routers/books.py
"""Book endpoints."""
from __future__ import annotations

from typing import List

from fastapi import APIRouter, Depends, status
from sqlalchemy.orm import Session

import crud
import schemas
from database import get_db


router = APIRouter(prefix="/books", tags=["books"])


@router.get("", response_model=List[schemas.BookOut])
def get_books(db: Session = Depends(get_db)):
    return crud.list_books(db)


@router.get("/{book_id}", response_model=schemas.BookOut)
def get_book(book_id: int, db: Session = Depends(get_db)):
    return crud.get_book(db, book_id)


@router.post("", response_model=schemas.BookOut, status_code=status.HTTP_201_CREATED)
def create_book(payload: schemas.BookCreate, db: Session = Depends(get_db)):
    return crud.create_book(db, payload)


@router.put("/{book_id}", response_model=schemas.BookOut)
def update_book(
    book_id: int, payload: schemas.BookUpdate, db: Session = Depends(get_db)
):
    return crud.update_book(db, book_id, payload)


@router.delete("/{book_id}", status_code=status.HTTP_204_NO_CONTENT)
def delete_book(book_id: int, db: Session = Depends(get_db)):
    crud.delete_book(db, book_id)
    return None
